dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/ServiceTrait.java

```diff
@@ -46,6 +46,10 @@ public boolean configure(Traits traitConfig, TraitContext context) {
             return traitConfig.getService().getEnabled();
         }
 
+        if (traitConfig.getContainer().getPort() != null) {
```

Comment:
I think we need similar for the Knative trait as it also checks on `TraitHelper.exposesHttpService()`